src/ray/object_manager/plasma/client.cc

```diff
@@ -751,8 +756,14 @@ Status PlasmaClient::Impl::EnsureGetAcquired(
   }
 
   int64_t version_read = 0;
+
+  // Need to unlock the client mutex since ReadAcquire() is blocking.
+  // TODO(ekl) is this entirely thread-safe?
```

Comment:
We are probably assured the object cannot be deallocated while a reader has the reference. @stephanie-wang @rkooo567 does this seem right?

Reply:
Right now mutable objects are never deallocated :D

But yeah in general this should be okay as long as we make sure to increment the PlasmaClient's local ref count for the object before we unlock.